modules/sample-ztp-custom-resources.adoc

```diff
@@ -34,6 +34,7 @@ You can customize the following {ztp} custom resources to specify more details a
       controlPlaneAgents: 1
       workerAgents: 0
     sshPublicKey: <YOUR_SSH_PUBLIC_KEY>
+    mastersSchedulable: true
```

Comment:
Whether or not this param would stay in this sample file in the long term, is it worth also adding a callout or note that explains how "this field enables workload scheduling of control planes at install"?

Reply:
I think the comment in the structure may help answer your question:

"MastersSchedulable allows masters nodes to be schedulable. When this flag is turned on, all the master nodes in the cluster will be made schedulable, so that workload pods can run on them. The default value for this field is false, meaning none of the master nodes are schedulable.
Important Note: Once the workload pods start running on the master nodes, extreme care must be taken to ensure that cluster-critical control plane components are not impacted."

Note also that if a cluster only has master nodes then has control planes nodes then by default MasterSchedulable will be set to 'true' if not defined in this manifest. Likewise, if a cluster has worker nodes defined then by MasterSchedulable will be set to 'false' unless set here.

Reply:
I think it's at least worth adding "Once the workload pods start running on the master nodes, extreme care must be taken to ensure that cluster-critical control plane components are not impacted." as some sort of IMPORTANT admonition under this sample.

Reply:
Per Bob on slack: "I'm not sure the installer doc is the place to add cluster specific monitoring information like this as we would have have to document how to detect the impact"

with that said, I won't add such a note unless this can be clarified and/or another SME/QE could confirm that it should be added

Reply:
For reference, here's how we used to describe how to do it on day 2:
https://docs.openshift.com/container-platform/4.7/nodes/nodes/nodes-nodes-working.html#nodes-nodes-working-master-schedulable_nodes-nodes-working
Those docs were removed in 4.8 though.

I can't find any mention in the OpenShift docs of how to do this on day 1 in the IPI/UPI install methods, even though it is possible. The only place I see it documented is in this KB: https://access.redhat.com/solutions/4685721

Maybe the correct answer is we should add these docs to that KB rather than to the product docs.

Reply:
> The default value for this field is false, meaning none of the master nodes are schedulable.

Note that in the agent installer the default value is configured correctly based on topology in the install-config, so for SNO/Compact clusters it will be true without the user needing to do anything.